packages/next/build/babel/preset.ts

```diff
@@ -71,7 +71,6 @@ module.exports = (api: any, options: NextBabelPresetOptions = {}): BabelPreset =
         development: isDevelopment || isTest,
         ...options['preset-react']
       }],
-      require('@babel/preset-typescript')
```

Comment:
This has to be part of the preset. The preset already uses overrides to detect `.ts` or `.tsx`, so there's no need to remove it.